File: domain/src/main/java/com/arturogutierrez/openticator/domain/account/model/Issuer.java
package com.arturogutierrez.openticator.domain.account.model;

public enum Issuer {

  UNKNOWN(Constants.ID_UNKNOWN),
  GOOGLE(Constants.ID_GOOGLE);

  private final String identifier;

  Issuer(String identifier) {
    this.identifier = identifier;
  }

  public String getIdentifier() {
    return identifier;
  }

  public static Issuer fromString(String identifier) {
    if (identifier == null) {
      return UNKNOWN;
    }

    if (identifier.equals(Constants.ID_GOOGLE)) {
      return GOOGLE;
    }

    return UNKNOWN;
  }

  private static class Constants {
    private static final String ID_UNKNOWN = "unknown";
    private static final String ID_GOOGLE = "google";
  }
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/issuer/model/Issuer.kt
package com.arturogutierrez.openticator.domain.issuer.model

enum class Issuer(val identifier: String) {

  UNKNOWN("unknown"),
  AWS("aws"),
  BITCOIN("bitcoin"),
  DIGITALOCEAN("digitalocean"),
  DROPBOX("dropbox"),
  EVERNOTE("evernote"),
  FACEBOOK("facebook"),
  GITHUB("github"),
  GOOGLE("google"),
  MICROSOFT("microsoft"),
  SLACK("slack"),
  ROBERTSPACEINDUSTRIES("rsi"),
  WORDPRESS("wordpress");

  companion object {
    private val map = Issuer.values().associateBy(Issuer::identifier)
    fun fromString(identifier: String): Issuer {
      val issuer = map[identifier]
      return issuer ?: UNKNOWN
    }
  }
}
